Userland/Libraries/LibGfx/ImageFormats/TinyVGLoader.cpp

```diff
@@ -510,7 +510,27 @@ struct TinyVGLoadingContext {
 static ErrorOr<void> decode_header_and_update_context(TinyVGLoadingContext& context)
 {
     VERIFY(context.state == TinyVGLoadingContext::State::NotDecoded);
+
+    auto expected_color_table_size = [](u32 color_count, ColorEncoding color_encoding) -> u64 {
+        switch (color_encoding) {
+        case ColorEncoding::RGBA8888:
+            return color_count * 4;
+        case ColorEncoding::RGB565:
+            return color_count * 2;
+        case ColorEncoding::RGBAF32:
+            return color_count * 16;
+        case ColorEncoding::Custom:
+            break;
+        }
+
+        VERIFY_NOT_REACHED();
+    };
+
     context.header = TRY(decode_tinyvg_header(context.stream));
+
+    if (context.stream.remaining() <= expected_color_table_size(context.header.color_count, context.header.color_encoding))
+        return Error::from_string_literal("Invalid TinyVG: Color table too large");
```

Comment:
nit: I think this error is a little imprecise for what this is actually checking.
```suggestion
        return Error::from_string_literal("Invalid TinyVG: Reported color table size larger than the remaining number of bytes");
```